### PR TITLE
problem with unionFind test, it returns all elements from array as "true", when expected digits

### PR DESCRIPTION
**Pull request by AlexDoh**
problem with unionFind test, it returns all elements from array as "true", when expected digits

### CODE REVIEW

File: entry-test/unionFind.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+        let componentsArray = 0;
+        let i, j;
+
+        for( i = 0, j = 0;i < this._elements.length;i++){
+            if(isNaN(this._elements[i])){
+                continue;
+            }
+
+            if(this._elements[i] = this._elements.indexOf(i)){
```
Comment (pavlom0x) [inline entry-test/unionFind.js:64]: are you sure?  `=` 
if you want to compare values, use `==` or `===`

Comment (pavlom0x) [reply entry-test/unionFind.js:64]: each time you call indexOf, it go through all array

please tell me what complexity(in Big O notation) of this method?

**Review by pavlom0x — CHANGES_REQUESTED** [review]

File: entry-test/unionFind.js (hunk 1 above)
Comment (AlexDoh) [reply entry-test/unionFind.js:64]: if it goes through all array - it is O(n).

Comment (AlexDoh) [reply entry-test/unionFind.js:64]: But I still don't understand why the array has only 'true' values...

**Pull request closed without merging**